### PR TITLE
UX: Admin styling

### PR DESCRIPTION
**Pull request by Chapoi**
image

image

### CODE REVIEW

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs [hunk 1]
```diff
@@ -52,9 +70,20 @@
       action=(action "saveEdit")
       disabled=saveEditDisabled
     }}
+    {{d-button
+      class="btn-secondary"
+      label="gamification.cancel"
+      title="gamification.cancel"
+      action=(action "saveEdit")
```
Comment (pmusaraj) [inline assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:77]: Should the action here be something like `cancelEdit`?

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs [hunk 2]
```diff
@@ -72,48 +101,81 @@
...
-                title="gamification.delete"
-                action=(action "destroyLeaderboard" leaderboard)
-              }}
-            </div>
-          </div>
-        </div>
-      {{/each}}
+      <table>
+        <thead>
+          <th>Name</th>
```
Comment (pmusaraj) [inline assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:124]: These labels need to be translatable.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs [hunk 3]
```diff
@@ -72,48 +101,81 @@
...
+          {{/each}}
+        </tbody>
+      </table>
     {{else}}
-      {{i18n "gamification.leaderboard.none"}}
+      {{#unless creatingNew}}
+        {{i18n "gamification.leaderboard.none"}}
+        {{d-button
+          label="gamification.leaderboard.cta"
+          title="gamification.leaderboard.cta"
```
Comment (pmusaraj) [inline assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:174]: I'm not sure it's useful to have identical label and title, screen readers will read the same text twice.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs [hunk 4]
```diff
@@ -1,49 +1,67 @@
...
-        value=selectedLeaderboard.name
-        placeholder=(i18n "gamification.leaderboard.name")
-      }}
-      <label class="control-label">{{i18n "gamification.leaderboard.date.range"}}</label>
-      <div class="controls">
-        {{date-picker-past
-          class="date-input"
-          value=selectedLeaderboard.from_date
-          onSelect=(action (mut fromDate))
+      <div class="leaderboard-admin__form-row">
```
Comment (pmusaraj) [inline assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:11]: This is part of a larger discussion, but I want to highlight an example of a consistency problem here. Core uses `control-group` for groupings of form elements. I know that core isn't super consistent across the different screens, that is something we should certainly try and fix. But that side, having a new class name in a plugin introduces even more inconsistency, any theme that styles `control-group`s won't be applying those styles to this particular grouping of form elements (and same for any future core stylesheet changes).

File: config/locales/client.en.yml [hunk 5]
```diff
@@ -13,16 +13,17 @@ en:
...
-          helper: "If dates are left empty the leaderboard will default to all available scores"
-        visible_to_groups: "Visible to Group(s)" 
-        visible_to_groups_help: "The group(s) that can view the the leaderboard" 
-        included_groups: "Included Group(s)" 
-        included_groups_help: "The group(s) that will be included in the leaderboard" 
+          helper: "If dates are left empty the leaderboard will show scores earned without any time restrictions."
+        visible_to_groups: "Visible to Group(s)"
+        visible_to_groups_help: "The group(s) that can view the the leaderboard."
+        included_groups: "Included Group(s)"
+        included_groups_help: "The group(s) that will be included in the leaderboard."
```
[the remark this answers is not in the document]
Comment (gschlager) [reply config/locales/client.en.yml:26]: I have no idea, so I've updated the suggestion. :grin:

File: assets/stylesheets/common/leaderboard-admin.scss [hunk 6]
```diff
@@ -0,0 +1,23 @@
+.leaderboard-admin {
+  &__title {
```
Comment (pmusaraj) [inline assets/stylesheets/common/leaderboard-admin.scss:2]: This `.leaderboard-admin__title` element is an H3, and core has default styles for headings in admin screens, I don't think plugins should be overriding them.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs (hunk 1 above)
Comment (Chapoi) [reply assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:77]: oops yeah huh, I def forgot to change this. thanks

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs (hunk 3 above)
Comment (Chapoi) [reply assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:174]: I just copied what the other buttons did tbh. Wasn't sure about changing it.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs (hunk 4 above)
Comment (Chapoi) [reply assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:11]: Ah ok so it has to be ```control-group``` then?

File: assets/stylesheets/common/leaderboard-admin.scss (hunk 6 above)
Comment (Chapoi) [reply assets/stylesheets/common/leaderboard-admin.scss:2]: True enough. It should be an h1 though actually.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs (hunk 3 above)
Comment (jjaffeux) [reply assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:174]: yes it's a mistake we made in the past, generally label alone is enough, unless you need more descriptive title, eg:

label: Add
title: Add Charlie to the designers group

File: assets/stylesheets/common/leaderboard-admin.scss (hunk 6 above)
Comment (pmusaraj) [reply assets/stylesheets/common/leaderboard-admin.scss:2]: Yeah, h1 would technically be the correct answer, except that other screens in the same route also use h3 (Core's own `/admin/plugins`, chat).

Comment (Chapoi) [reply assets/stylesheets/common/leaderboard-admin.scss:2]: Okay, that's part of a larger issue then. We should think about starting to fix these html/accessibility mistakes then gradually.

File: assets/javascripts/discourse/templates/admin-plugins-gamification.hbs [hunk 7]
```diff
@@ -72,48 +98,80 @@
...
-              }}
-            </div>
-          </div>
-        </div>
-      {{/each}}
+      <table>
+        <thead>
+          <th>{{i18n "gamification.admin.name"}}</th>
+          <th>{{i18n "gamification.admin.period"}}</th>
+          {{!-- <th>Contenders</th>
```
[the remark this answers is not in the document]
Comment (pmusaraj) [reply assets/javascripts/discourse/templates/admin-plugins-gamification.hbs:123]: No worries, code is not a good place for that sort of reminder, best to add it to gamification roadmap (on meta or internally). If left in code, it'll be there for a long time.

**Pull request merged**